enterprise/server/crypter_service/crypter_service.go

```diff
@@ -548,6 +548,7 @@ func (d *Decryptor) Read(p []byte) (n int, err error) {
 		// ErrUnexpectedEOF indicates that the underlying reader returned EOF
 		// before the buffer could be filled, which is expected on the last
 		// chunk.
+		lastChunk := err == io.ErrUnexpectedEOF
```

Comment:
Could there be an edge case where the reader perfectly fills the buffer on the read of the final chunk and doesn't return EOF? Would that prevent the final chunk from being read correctly?

Reply:
Yes, good point. A file written normally would always have a non-full chunk at the end, but we also want to make sure that the file cannot be truncated after the fact. I've added a check to make sure we actually come across a "last chunk" chunk before we finish with the reader.